code/modules/mob/status_procs.dm

```diff
@@ -26,66 +26,40 @@
 	if(amount>0)
 		var/old_eye_blind = eye_blind
 		eye_blind = max(eye_blind, amount)
-		if(!old_eye_blind)
-			if(stat == CONSCIOUS || stat == SOFT_CRIT)
-				throw_alert("blind", /obj/screen/alert/blind)
-			overlay_fullscreen("blind", /obj/screen/fullscreen/blind)
-			// You are blind why should you be able to make out details like color, only shapes near you
-			add_client_colour(/datum/client_colour/monochrome/blind)
+		if(!old_eye_blind || !eye_blind && !HAS_TRAIT(src, TRAIT_BLIND))
+			update_blindness()
 
 /**
   * Adjust a mobs blindness by an amount
   *
   * Will apply the blind alerts if needed
   */
 /mob/proc/adjust_blindness(amount)
-	if(amount>0)
-		var/old_eye_blind = eye_blind
-		eye_blind += amount
-		if(!old_eye_blind)
-			if(stat == CONSCIOUS || stat == SOFT_CRIT)
-				throw_alert("blind", /obj/screen/alert/blind)
-			overlay_fullscreen("blind", /obj/screen/fullscreen/blind)
-			add_client_colour(/datum/client_colour/monochrome/blind)
-	else if(eye_blind)
-		var/blind_minimum = 0
-		if((stat != CONSCIOUS && stat != SOFT_CRIT))
-			blind_minimum = 1
-		if(isliving(src))
-			var/mob/living/L = src
-			if(HAS_TRAIT(L, TRAIT_BLIND))
-				blind_minimum = 1
-		eye_blind = max(eye_blind+amount, blind_minimum)
-		if(!eye_blind)
-			clear_alert("blind")
-			clear_fullscreen("blind")
-			remove_client_colour(/datum/client_colour/monochrome/blind)
+	var/old_eye_blind = eye_blind
+	eye_blind += amount
+	if(!old_eye_blind || !eye_blind && !HAS_TRAIT(src, TRAIT_BLIND))
+		update_blindness()
 /**
```

Comment:
this is effectively the same proc as the other one, just have one call the other